frontend/src/components/authentication/types.tsx

```diff
@@ -0,0 +1,20 @@
+import { NavigateFunction } from "react-router-dom";
+import { userSignedIn } from "../../contexts/app-context/Types";
+
+export interface UseMutateUserArgs {
+    URL: string;
+    setUserSignedIn: React.Dispatch<React.SetStateAction<userSignedIn | null>>;
+    navigate: NavigateFunction;
+    redirectionRoute: string;
+}
+export type SignUpFormProps = {
+    handleUserSignUpSubmit: (event: React.FormEvent<HTMLFormElement>) => Promise<void>;
+    handleNavigate: () => void;
+};
+
+export type UserData = {
+    name?: string;
```

Comment:
why name is optional?

Reply:
So that i use this type in both usePostSignInUser and usePostSignUpUser, in usePostSignInUser name is not required only in  
usePostSignUpUser, do you think it is better if i make a type for each function instead?
it will be like this :
`export type UserSignUpData = {
    name: string;
    email: string;
    password: string;
    confirmPassword: string;
};`

`export type UserSignInData = {
    email: string;
    password: string;
};`

Reply:
Yess, it's better to separate them 👌